### PR TITLE
pipeline: fully move to S3

### PR DESCRIPTION
**Pull request by jlebon**
Now that we have a prod build in S3, we can fully switch over to it.
Most of the changes are straightforward. One note of interest: the
"pruning" stage is now really just about pruning the local cache. For
now, we're not pruning from the bucket at all, pending a more defined
policy & mechanism.

### CODE REVIEW

**Comment by jlebon** [comment]
(Still testing this.)

**Comment by jlebon** [comment]
OK, dropping WIP, now tested! :heavy_check_mark:

File: Jenkinsfile [hunk 1]
```diff
@@ -102,18 +102,12 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
         }
 
         stage('Fetch') {
-            // XXX: drop `!prod && ` once we've uploaded prod builds there
-            if (!prod && s3_builddir) {
```
Comment (dustymabe) [inline Jenkinsfile]: so we ran this once already? or do we need https://github.com/coreos/coreos-assembler/pull/545 first?

**Comment by dustymabe** [comment]
this mostly LGTM. only concern I have is if we (or anyone else) want to set up another pipeline and not touch s3. Would keeping the rsync functionality around be useful ?

File: Jenkinsfile (hunk 1 above)
Comment (jlebon) [reply Jenkinsfile]: We did `buildupload` once already, so we have something to `buildprep` from. https://github.com/coreos/coreos-assembler/pull/545 is for the case where there's nothing to `buildprep` from (which is mostly relevant for devel pipelines).

**Comment by jlebon** [comment]
OK, I've reworked this for now so that we still do upload the latest build to the artifact server. From comment:

```groovy
// XXX: For now, we keep uploading the latest build to the artifact
// server to make it easier for folks to access since we don't have
// a stream metadata frontend/website set up yet. The key part here
// is that it is *not* the canonical storage for builds.
```

**Comment by jlebon** [comment]
> Would keeping the rsync functionality around be useful ?

So as per the previous comment, I've kept it for now for another reason. Though once we're ready to move on for good, I think I'd rather we drop it completely until someone actually shows up with that use case. `git` will be there, remembering everything.

Note for the no-S3 devel pipeline, we do store builds in the PVC. So one should be able to just bring up e.g. simple-httpd to access it over HTTP if they'd like.

**Comment by dustymabe** [comment]
> OK, I've reworked this for now so that we still do upload the latest build to the artifact server. From comment:
> 
> ```groovy
> // XXX: For now, we keep uploading the latest build to the artifact
> // server to make it easier for folks to access since we don't have
> // a stream metadata frontend/website set up yet. The key part here
> // is that it is *not* the canonical storage for builds.
> ```

hmm i'd prefer to just open up the bucket and make it brows-able IMHO



> So as per the previous comment, I've kept it for now for another reason. Though once we're ready to move on for good, I think I'd rather we drop it completely until someone actually shows up with that use case. `git` will be there, remembering everything.

I'm OK with letting `git` store it 👍

**Comment by jlebon** [comment]
> hmm i'd prefer to just open up the bucket and make it brows-able IMHO

I thought consensus was to *not* have it browsable, right? Because we don't want users mucking around in there in the first place. See e.g. https://github.com/coreos/fedora-coreos-tracker/issues/169#issuecomment-497522046.

**Comment by dustymabe** [comment]
> > hmm i'd prefer to just open up the bucket and make it brows-able IMHO
> 
> I thought consensus was to _not_ have it browsable, right? Because we don't want users mucking around in there in the first place. See e.g. [coreos/fedora-coreos-tracker#169 (comment)](https://github.com/coreos/fedora-coreos-tracker/issues/169#issuecomment-497522046).

yeah should have clarified *make it brows-able until we get a release browswer*

either way I guess we can still rsync the content out and just point people there for now.

**Comment by jlebon** [comment]
AFAIK, S3 doesn't have a native "web browser" (like e.g. the Apache listing). Making the bucket "browsable" would mean indexing it with something like https://github.com/projectatomic/papr/blob/master/papr/utils/indexer.py which drops a bunch of `index.html` files everywhere. :grimacing:

**Pull request merged**